src/main/java/org/folio/oaipmh/service/impl/SetServiceImpl.java

```diff
@@ -1,25 +1,58 @@
 package org.folio.oaipmh.service.impl;
 
+import static javax.ws.rs.core.HttpHeaders.ACCEPT;
+import static javax.ws.rs.core.MediaType.APPLICATION_JSON;
+import static org.folio.oaipmh.Constants.OKAPI_TENANT;
+import static org.folio.oaipmh.Constants.OKAPI_TOKEN;
+import static org.folio.oaipmh.Constants.OKAPI_URL;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
 import org.folio.oaipmh.dao.SetDao;
 import org.folio.oaipmh.service.SetService;
+import org.folio.rest.jaxrs.model.FilteringConditionValueCollection;
 import org.folio.rest.jaxrs.model.Set;
+import org.folio.rest.jaxrs.model.SetsFilteringCondition;
 import org.springframework.stereotype.Service;
 
+import io.vertx.core.CompositeFuture;
 import io.vertx.core.Future;
-import io.vertx.core.logging.Logger;
-import io.vertx.core.logging.LoggerFactory;
+import io.vertx.core.Promise;
+import io.vertx.core.Vertx;
+import io.vertx.core.http.HttpClient;
+import io.vertx.core.http.HttpClientRequest;
+import io.vertx.core.json.JsonObject;
 
 @Service
 public class SetServiceImpl implements SetService {
 
-  private static final Logger logger = LoggerFactory.getLogger(SetServiceImpl.class);
+  private static final String LOCATION_URI = "/locations";
+  private static final String ILL_POLICIES_URI = "/ill-policies";
+  private static final String MATERIAL_TYPES_URI = "/material-types";
+  private static final String RESOURCE_TYPES_URI = "/instance-types";
+  private static final String INSTANCE_FORMATS_URI = "/instance-formats";
+
+  private static final String LOCATION_JSON_FIELD_PATH = "locations";
+  private static final String ILL_POLICIES_JSON_FIELD_PATH = "illPolicies";
+  private static final String MATERIAL_TYPES_JSON_FIELD_PATH = "mtypes";
+  private static final String INSTANCE_TYPES_JSON_FIELD_PATH = "instanceTypes";
+  private static final String INSTANCE_FORMATS_JSON_FIELD_PATH = "instanceFormats";
+
+  private static final String LOCATION = "location";
```

Comment:
I don't understand why these need to be restricted to private... it means you need to redefine them again in your test(s).  IMO it's better to open them up to protected, or even public if needed and then use static imports in your test(s)

Reply:
> I don't understand why these need to be restricted to private... it means you need to redefine them again in your test(s). IMO it's better to open them up to protected, or even public if needed and then use static imports in your test(s)
Strongly agree with you, I will fix it now

Reply:
Well this creates a dependency, unwanted from my perspective, to the code from tests. My preference would be either we  move this to Constancts or use plain strings in tests...

Reply:
These are unit tests aren't they?  Isn't there an inherent dependency (even if via the indirection of API) on the implementation from the test anyway?

Regardless, the point is, repeating the same thing over and over again is a PITA to maintain... if you want to change it, you need to change it in N places.

Moving these into a Constants class is an acceptable compromise.